dist/tools/usb-serial/ttys.py

```diff
@@ -0,0 +1,132 @@
+        if tty.serial == forbidden_serial:
+            return False
+
+    return True
+
+
+def shorten(string, length):
+    """
+    Shorten the given string to the given length, if needed
+    """
+    if len(string) > length:
+        return string[:length - 3] + "..."
+
+    return string
+
+
+def print_ttys(args):
+    """
+    Print ttys as specified by the given command line arguments
+    """
+    desc = "List and filter TTY interfaces that might belong to boards"
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument("--most-recent", action="store_true",
+                        help="Print only the most recently connected matching "
+                             + "TTY")
+    parser.add_argument("--path-only", action="store_true",
+                        help="Print only the device path of the matches")
+    parser.add_argument("--serial", default=None, type=str,
+                        help="Print only devices matching this serial")
+    parser.add_argument("--driver", default=None, type=str,
+                        help="Print only devices using this driver")
+    parser.add_argument("--model", default=None, type=str,
+                        help="Print only devices matching this model")
+    parser.add_argument("--vendor", default=None, type=str,
+                        help="Print only devices matching this vendor")
+    parser.add_argument("--exclude-serial", type=str, nargs='*', default=None,
+                        help="Ignore devices with these serial numbers. "
+                             + "Environment variable EXCLUDE_TTY_SERIAL can "
+                             + "be used alternatively.")
+
+    args = parser.parse_args()
+    if args.exclude_serial is None:
+        if "EXCLUDE_TTY_SERIAL" in os.environ:
+            args.exclude_serial = os.environ["EXCLUDE_TTY_SERIAL"].split()
+        else:
+            args.exclude_serial = []
+
+    ttys = []
+    for dev in pyudev.Context().list_devices(subsystem='tty', ID_BUS='usb'):
+        tty = Tty(dev)
+        if filters_match(args, tty):
+            ttys.append(tty)
+
+    if args.most_recent:
+        most_recent = ttys[0]
+        for tty in ttys:
+            if tty.ctime > most_recent.ctime:
+                most_recent = tty
+        ttys = [most_recent]
+
+    if not args.path_only:
+        print(f"{'Path':13} | {'Driver':10} | {'Vendor':20} | {'Model':30} "
+              f"| {'Serial':30} | {'ctime':8}")
+        print("-" * 13 + "-|-" + "-" * 10 + "-|-" + "-" * 20 + "-|-"
+              + "-" * 30 + "-|-" + "-" * 30 + "-|-" + "-" * 8)
+
+    for tty in ttys:
+        if args.path_only:
+            print(tty.path)
+        else:
+            ctime = time.strftime("%H:%M:%S", time.localtime(tty.ctime))
+            vendor = shorten(tty.vendor, 20)
+            model = shorten(tty.model, 30)
+            print(f"{tty.path:13} | {tty.driver:10} | {vendor:20} "
+                  f"| {model:30} | {tty.serial:30} | {ctime:8}")
```

Comment:
Not sure how to best handle long serials (see example in my previous post) -- shortening them would make the table work again, but then how does one get it out (or the model for `--model`, for that matter)?

Reply:
I guess I just increase the column width. (I think shortening the serial would be non-ideal, as we might refer to `make list-ttys` to retrieve the serial.)

I will later check if I can find my micro:bit and see if I can the correct udev key to retrieve that. I fully agree that this is the string one is looking for when plugging in a `micro:bit`.